modules/bmp/image_loader_bmp.cpp

```diff
@@ -91,11 +91,15 @@ Error ImageLoaderBMP::convert_to_image(Ref<Image> p_image,
 		// the data width in case of 8/4/1 bit images
 		const uint32_t w = bits_per_pixel >= 24 ? width : width_bytes;
 		const uint8_t *line = p_buffer + (line_width * (height - 1));
+		const uint8_t *end_buffer = p_buffer + p_header.bmp_info_header.bmp_size_image;
 
 		for (uint64_t i = 0; i < height; i++) {
 			const uint8_t *line_ptr = line;
 
 			for (unsigned int j = 0; j < w; j++) {
+				if (line_ptr >= end_buffer) {
+					return ERR_FILE_CORRUPT;
```

Comment:
We use ERR_FAIL_COND typically instead of an if statement because one extra thing it does is show the line number.